subsys/bluetooth/scan.c

```diff
@@ -20,6 +20,8 @@ LOG_MODULE_REGISTER(nrf_bt_scan, CONFIG_BT_SCAN_LOG_LEVEL);
 
 /* Scan filter add mutex. */
 K_MUTEX_DEFINE(scan_add_mutex);
+K_MUTEX_DEFINE(conn_attempts_mutex);
```

Comment:
I am not sure so many locks are needed.
Note that each one will use RAM (+ ROM + cause cache pollution if cache is used).

One should be enough as I think both lower (stack) and upper (application) layer will use this module from the same context. I don't think there will be any savings from using 3 separate locks.

Reply:
I think that it could be fair enough to have only one mutex. Done